microbench/src/main/java/io/netty/handler/codec/http/DecodeHexBenchmark.java

```diff
@@ -43,37 +47,71 @@
             "4DDeA5gDD1C6fE567E1b6gf0C40FEcDg",
     })
     private String hex;
-    private char[] hexDigits;
+    @Param({ "1", "10" })
+    private int inputs;
+    private char[][] hexDigits;
+    private static final long SEED = 1578675524L;
 
     @Setup
     public void init() {
-        hexDigits = hex.toCharArray();
+        final char[] hexCh = hex.toCharArray();
+        hexDigits = new char[inputs][];
+        hexDigits[0] = hexCh;
+        if (inputs > 1) {
+            final Random rnd = new Random(SEED);
+            for (int i = 1; i < inputs; i++) {
+                hexDigits[i] = shuffle(Arrays.copyOf(hexCh, hexCh.length), rnd);
+            }
+        }
+    }
+
+    // https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle
+    private static char[] shuffle(char[] chars, Random rnd) {
+        int index;
+        char tmp;
+        for (int i = chars.length - 1; i > 0; i--) {
+            index = rnd.nextInt(i + 1);
+            tmp = chars[index];
+            chars[index] = chars[i];
+            chars[i] = tmp;
+        }
+        return chars;
+    }
+
+    private int nextHexDigits() {
+        // always use ThreadLocalRandom here:
+        // we want the inputs = 1 and inputs > 1 to be comparable
+        // and saving using ThreadLocalRandom have impacts!
+        return ThreadLocalRandom.current().nextInt(0, inputs);
     }
 
     @Benchmark
+    @CompilerControl(Mode.DONT_INLINE)
     public long hexDigits() {
         long v = 0;
```

Comment:
why long not int? and I guess something like `^=` might be cheaper than `+=`? (maybe insignificant though...)

Reply:
>  (maybe insignificant though...)

Just tried and on my laptop not getting any meaningful differences!